setup.py

```diff
@@ -153,6 +162,18 @@ def run(self):
             except ImportError:
                 pass
 
+    if os.environ.get("PYOGRIO_PACKAGE_DATA"):
```

Comment:
I think these will fail if GDAL is a valid install (e.g., on MacOS / Linux) , but env vars are not set?

Suggest a little restructuring here, and raise exceptions if the paths are not found: `PYOGRIO_PACKAGE_DATA` signals intent that we expect to package these, so we should hard fail if we can't.

Untested code:

```Python
gdal_data = os.environ.get("GDAL_DATA")
if not gdal_data:
    gdal_data = find_gdal_data()
if not gdal_data and os.path.exists(gdal_data):
    raise Exception("Could not find GDAL data files for packaging")

proj_data = os.environ.get("PROJ_LIB")
if not proj_data:
    proj_data = find_proj_data()
if not proj_data and os.path.exists(proj_data):
    raise Exception("Could not find PROJ data files for packaging")    

log.info(f"Copying GDAL data from {gdal_data} and PROJ data from {proj_data}")

copy_data_tree(gdal_data, "pyogrio/gdal_data")
copy_data_tree(proj_data, "pyogrio/proj_data")
package_data = {"pyogrio": ["gdal_data/*", "proj_data/*"]}


def find_gdal_data():
    gdal_config = os.environ.get("GDAL_CONFIG", "gdal-config")
    return read_response([gdal_config, "--datadir"])
   
def find_proj_data():
    return read_response(['projinfo', "--searchpaths"]).split('\n')[-1]
```

`find_proj_data` is arguably less robost (`find_gdal_data` reuses things we already check), so it is reasonable to expect us to set `PROJ_LIB` on platforms where that is in a nonstandard location or not easily found.

Alternatively, if you find that falling back to finding them too complex, it is not unreasonable for us to insist that `GDAL_DATA` and `PROJ_LIB` be set in order to create wheels.

Reply:
We could also leave this extra logic until it is needed / requested? Right now, we only use this for wheel building and on windows, in which case those env variables are always set. Even for wheel building on linux/mac, it might be that we end up defining those variables. 

I suppose this would mostly be for external people who would want to build wheels themselves? (although I would say that it doesn't seem such a burden to ask them to define those variables, _if_ they want to include data. They would need to set the PYOGRIO_PACKAGE_DATA flag anyway)

But I can certainly change it to error if they are not set.

Reply:
> We could also leave this extra logic until it is needed / requested?

It is fine to leave it for now.